openedx/core/djangoapps/oauth_dispatch/docs/decisions/0007-include-organization-in-tokens.rst

```diff
@@ -0,0 +1,192 @@
+7. Include Organization in Tokens
+---------------------------------
+
+Status
+------
+
+Proposed
+
+Context
+-------
+
+Status of Organizational Access to edX APIs
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+External edX applications would like to make server-to-server API
+calls via the Client Credentials grant type to access data. However,
+our APIs typically return data only to global staff users who
+effectively have administrative read access to the system. This
+all-or-nothing capability is unsatisfactory to meet the needs of
+edX partner organizations.
+
+Additionally, some organizations create their own web portals for
+their learners, using edX as an identity provider and as the underlying
+LMS. For various reasons (?), they would like to present edX data to
+their learners on their own portal. Currently, they cannot access a
+learner's data using our APIs.
+
+Our API endpoints do not have more flexible capabilities since they
+do not have reliably sufficient information to limit/filter API results
+according to the organizational affiliation of the requesting application.
+
+Organizational Types in the edX System
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In the edX system, the 2 most prevalent organizational relationships
+are:
+
+* **Organization as a Content Provider**
+
+  * This is a partner organization that provides content for a course,
+    program, etc. Typically, such an organization will want to access
+    data for all learners enrolled in their courses. They may choose to
+    do so either via
+
+    a. bulk APIs using the *Client Credentials grant type* (e.g., to
+       synchronize their own data in a background process) or 
+
+    b. a user-specific API on behalf of a logged-in user via the 
+       *Authorization grant type* and *edX as the identity provider*
+       (e.g., to display user-specific data on their own portal).
+
+* **Organization as a User Provider**
+
+  * This is an enterprise organization that registers users onto the
+    edX system, typically via an SSO-enabled portal, but with the
+    *organization (not edX) as the identity provider*. Such an
+    organization will also want to access data for all its users.
+    However, it is not an immediate requirement to support data
+    access by this organization type at this time.
```

Comment:
Why not? FYI @georgebabey

Reply:
+1 This is functionality we need for the data api work

Reply:
In that case, I'd like to work with your team to create a separate 0008 decision regarding how we would handle this for Enterprise organizations.  I don't believe this is a requirement for MS - and so they may not include it in their work.  However, I will confirm with them at a Feb 15th meeting.